### DISCUSSION TITLE
Workflow for automatically commit and push changes in source directory

### DISCUSSION THREAD

**Discussion by axhamre** [twpayne/chezmoi#2738] [Q&A] [answered]
When I add or re-add files they get automatically committed and pushed to my repo.

However for daily changes this doesn't seem to work for me, i.e. when I do changes in my source directory and then run `chezmoi apply`. Is this supposed to work?

My chezmoi.toml:

```
[edit]
    command = "code"
    args = ["--wait"]
[git]
    autoCommit = true
    autoPush = true
```

**Comment by twpayne** [comment, marked as answer]
chezmoi 

> However for daily changes this doesn't seem to work for me, i.e. when I do changes in my source directory and then run `chezmoi apply`. Is this supposed to work?

No. `autoCommit` and `autoPush` only work when chezmoi knows that you've made changes to your source directory, i.e. after commands like `chezmoi add`, `chezmoi edit`, `chezmoi import`, and so on.

`chezmoi apply` only tells chezmoi to update your home directory. You might have local changes that you're just testing and don't want to commit yet, so it doesn't autocommit or autopush.
